File: ProductCatalog/src/main/java/com/Ecommerce/ProductCatalog/Model/Product.java
package com.Ecommerce.ProductCatalog.Model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "products")
public class Product {
    @Id
    private String id;

    private String unique_id;
    private String URL;
    private String name;
    private List<String> category;
    private String description;
    private String brand;
    private int retailPrice;
    private int discountedPrice;
    private List<String> images;
    private int stock;
}

File: SearchEngine/SearchEngine/Models/Product.cs
using System;
using System.Collections.Generic;

namespace SearchEngine.Models
{
    public class Product
    {
        public Guid Id { get; set; }
        public string UniqueId { get; set; }
        public string Url { get; set; }
        public string Name { get; set; }
        public List<string> Category { get; set; }
        public string Description { get; set; }
        public string Brand { get; set; }
        public float RetailPrice { get; set; }
        public float DiscountedPrice { get; set; }
        public List<string> Images { get; set; }
        public int Stock { get; set; }
    }
}